ci_tools/bot_interaction.py

```diff
@@ -217,13 +217,23 @@ def mark_as_ready(pr_id):
 
     job_data = [j for j in data if j['name'] not in ('Bot', 'CleanUpBot')]
 
-    failures = [j['name'] for j in job_data if j['conclusion'] in ('cancelled', 'failed')]
+    failures = [j.get('name',"") for j in job_data if j['conclusion'] in ('cancelled', 'failure')]
 
     if any(j['name'] == 'Codacy Static Code Analysis' and j['conclusion'] in ('FAILURE', 'ACTION_REQUIRED') for j in running_job_data):
```

Comment:
When you created the list `job_data` just above, you used `j['name']`, therefore every `j` dictionary in the list has the key `name`. (If not, a `KeyError` exception would have been raised.) So why do you have to use `j.get('name', "")` to build the list `failures`?

Reply:
I made this change as I saw a `KeyError` in the output of a test at one point (I'm not sure exactly what leads to a nameless test). I must have got the lines mixed up. I have fixed it : [771eed8](https://github.com/pyccel/pyccel/pull/1369/commits/771eed8fd8fc09f0d89790fa347c65552ab3db16)